src/iidm/converter/xml/VoltageLevelXml.cpp

```diff
@@ -48,6 +56,58 @@ const char* VoltageLevelXml::getRootElementName() const {
     return VOLTAGE_LEVEL;
 }
 
+void VoltageLevelXml::readBusBreakerTopology(VoltageLevel& voltageLevel, NetworkXmlReaderContext& context) const {
+    context.getReader().readUntilEndElement(BUS_BREAKER_TOPOLOGY, [&voltageLevel, &context]() {
+        if (context.getReader().getLocalName() == BUS) {
+            BusXml::getInstance().read(voltageLevel, context);
+        } else if (context.getReader().getLocalName() == SWITCH) {
+            BusBreakerViewSwitchXml::getInstance().read(voltageLevel, context);
+        } else {
+            throw powsybl::xml::XmlStreamException(logging::format("Unsupported element %1%", context.getReader().getLocalName()));
+        }
+    });
+}
+
+void VoltageLevelXml::readCalculatedBus(VoltageLevel &voltageLevel, NetworkXmlReaderContext& context) const {
+    xml::IidmXmlUtil::assertMinimumVersion(VOLTAGE_LEVEL, BUS, xml::ErrorMessage::NOT_SUPPORTED, xml::IidmXmlVersion::V1_1(), context.getVersion());
+    double v = context.getReader().getOptionalAttributeValue(V, stdcxx::nan());
+    double angle = context.getReader().getOptionalAttributeValue(ANGLE, stdcxx::nan());
+    const std::string& nodesString = context.getReader().getAttributeValue(NODES);
+    context.addEndTask([angle, nodesString, &voltageLevel, v]() {
```

Comment:
pass the parameters in this order: [v, angle, &nodeString, &voltagelLevel]
- V and Angle are together
- nodeString is passed by address

Reply:
Ok for the capture order.
But concerning `nodeString`, it cannot be passed by address because it would have deleted when the end task function will be called. Do you prefer using shared_ptr instead instead of passing by copy ?